qa/sc_evm_gasPrice.py

```diff
@@ -0,0 +1,111 @@
+    - Create more blocks to get out of 40 block range and check that gasPrice = baseFee
+    - Create txs that have 600 GWei as maxPriorityFeePerGas
+    - Create a block and check that gasPrice returns maxPrice of 500 GWei
+     
+"""
+
+
+class SCEvmGasPrice(AccountChainSetup):
+    def __init__(self):
+        super().__init__(withdrawalEpochLength=80)
+
+    def __do_send_raw_tx(self, raw_tx, evm_signer_address):
+        sc_node = self.sc_nodes[0]
+        signed_raw_tx = signTransaction(sc_node, fromAddress=evm_signer_address, payload=raw_tx)
+
+        tx_hash = sendTransaction(sc_node, payload=signed_raw_tx)
+        return tx_hash
+
+    def __do_check_gas_price(self, maxPriorityFeePerGas, generate_blocks=1):
+        maxPrice = 500000000000
+        sc_node = self.sc_nodes[0]
+
+        if generate_blocks > 0:
+            generate_next_blocks(sc_node, "first node", generate_blocks)
+
+        expected_price = int(sc_node.rpc_eth_getBlockByNumber('latest', True)['result']['baseFeePerGas'], 16)
+        if (maxPriorityFeePerGas + expected_price) > maxPrice:
+            expected_price = maxPrice
+        else:
+            expected_price += maxPriorityFeePerGas
+
+        assert_true(int(sc_node.rpc_eth_gasPrice()['result'], 16) == expected_price)
+
+    def run_test(self):
+        ft_amount_in_zen = Decimal('1000.0')
+
+        self.sc_ac_setup(ft_amount_in_zen=ft_amount_in_zen)
+        sc_node = self.sc_nodes[0]
+
+        evm_address_1 = remove_0x_prefix(self.evm_address)
+        evm_address_2 = sc_node.wallet_createPrivateKeySecp256k1()["result"]["proposition"]["address"]
+
+        # no txs available - return current base fee
+        self.__do_check_gas_price(maxPriorityFeePerGas=0, generate_blocks=1)
+
+        nonce_addr_1 = 0
+        maxPriorityFeePerGas = 1500000
+        transferred_amount_in_zen = Decimal('1.2')
+
+        # Create some txs for gasPrice calculation later
+        for i in range(5):
+            raw_tx = createRawEIP1559Transaction(sc_node,
+                                                 fromAddress=evm_address_1,
+                                                 toAddress=evm_address_2,
+                                                 value=convertZenToWei(transferred_amount_in_zen),
+                                                 nonce=str(nonce_addr_1),
+                                                 maxPriorityFeePerGas=maxPriorityFeePerGas)
+            self.__do_send_raw_tx(raw_tx=raw_tx, evm_signer_address=evm_address_1)
+            nonce_addr_1 += 1
+
+        generate_next_block(sc_node, "first node")
+        assert_true(len(sc_node.rpc_eth_getBlockByNumber('latest', False)['result']['transactions']) == 5)
+
+        self.__do_check_gas_price(maxPriorityFeePerGas, generate_blocks=10)
+
+        self.__do_check_gas_price(maxPriorityFeePerGas, generate_blocks=15)
+
+        self.__do_check_gas_price(maxPriorityFeePerGas=0, generate_blocks=20)
```

Comment:
This is a bit confusing to read, as there is both a parameter and a local variable called `maxPriorityFeePerGas`. It looks like it is setting the local variable to `0`, but it is actually passing the parameter as `0`.

Reply:
I get what you mean - I just explicitly wrote it like this, because I wanted to show - in the call - that the function is called with 0 as a value for maxPriorityFeePerGas.
Should I remove the `maxPriorityFeePerGas=` or just rename the local variable?